src/it/scala/com/exasol/cloudetl/kinesis/KinesisAbstractIntegrationTest.scala

```diff
@@ -0,0 +1,178 @@
+    print(files)
+    val jarFile = files.find(_.getName.contains(name))
+    jarFile match {
+      case Some(jarFilename) => jarFilename.getName
+      case None =>
+        throw new IllegalArgumentException(
+          "Cannot find a file or a directory with pattern" + name + " in " + directoryToSearch
+        )
+    }
+  }
+
+  private[kinesis] def createKinesisStream(streamName: String, shardsCounter: Integer): Unit = {
+    kinesisClient.createStream(streamName, shardsCounter)
+    // We have to wait until stream is ready to be accessed.
+    Thread.sleep(30 * 1000)
+  }
+
+  def findAssembledJarName(): String = {
+    val scalaDirectory = getScalaDirectory()
+    findFileOrDirectory("target/" + scalaDirectory, JAR_NAME_PATTERN)
+  }
+
+  private[this] def listDirectoryFiles(directoryName: String): List[File] = {
+    val directory = new File(directoryName)
+    if (directory.exists && directory.isDirectory) {
+      directory.listFiles.toList
+    } else {
+      List.empty[File]
+    }
+  }
+
+  private[kinesis] def putRecordIntoStream(
+    sensorId: Int,
+    currentTemperature: Int,
+    status: String,
+    partitionKey: String,
+    streamName: String
+  ): Unit = {
+    val recordData =
+      s"""{\"sensorId\": $sensorId,
+         | \"currentTemperature\": $currentTemperature,
+         | \"status\": \"$status\"
+         | }""".stripMargin.replace("\n", "")
+    val data = ByteBuffer.wrap(recordData.getBytes())
+    kinesisClient.putRecord(streamName, data, partitionKey)
+    ()
+  }
+
+  private[kinesis] def createKinesisMetadataScript(): Unit = {
+    statement.execute(
+      s"""CREATE OR REPLACE JAVA SET SCRIPT KINESIS_METADATA (...)
+         |EMITS (KINESIS_SHARD_ID VARCHAR(130), SHARD_SEQUENCE_NUMBER VARCHAR(2000)) AS
+         |     %jvmoption -Dcom.amazonaws.sdk.disableCbor=true;
+         |     %scriptclass com.exasol.cloudetl.kinesis.KinesisShardsMetadataReader;
+         |     %jar /buckets/bfsdefault/default/$assembledJarName;
+         |/
+         |""".stripMargin
+    )
+    ()
+  }
+
+  private[kinesis] def createKinesisImportScript(emits: String): Unit = {
+    statement.execute(
+      s"""CREATE OR REPLACE JAVA SET SCRIPT KINESIS_IMPORT (...)
+         |EMITS ($emits) AS
+         |     %jvmoption -Dcom.amazonaws.sdk.disableCbor=true;
+         |     %scriptclass com.exasol.cloudetl.kinesis.KinesisShardDataImporter;
+         |     %jar /buckets/bfsdefault/default/$assembledJarName;
+         |/
+         |""".stripMargin
+    )
+    ()
+  }
+
+  private[kinesis] def extractTuple(resultSet: ResultSet): (Int, Int, String, String, Boolean) =
+    (
+      resultSet.getInt("sensorId"),
+      resultSet.getInt("currentTemperature"),
+      resultSet.getString("status"),
+      resultSet.getString(KINESIS_SHARD_ID_COLUMN_NAME),
+      resultSet.getString(SHARD_SEQUENCE_NUMBER_COLUMN_NAME) != null
+    )
+
+  private[kinesis] def extractRow(resultSet: ResultSet): Row = {
```

Comment:
This function can be removed since tuple version is used.
Remove the `Row` import.

Reply:
done